src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -66,17 +68,63 @@ public function indexAction(WebserviceKeyFilters $filters, Request $request)
 
         $configurationWarnings = $this->lookForWarnings();
 
+        $webserviceConfiguration = $this->get('prestashop.admin.webservice.form_data_provider')->getData();
+        $webserviceIsEnabled = $webserviceConfiguration['webservice_configuration']['enable_webservice'];
+        $webserviceStatus = [
+            'enabled' => $webserviceIsEnabled,
+            'isFunctional' => true,
+            'endpoint' => null,
+        ];
+
+        if ($webserviceIsEnabled) {
+            $expectedWebserviceEntryPoint = $request->getSchemeAndHttpHost() . self::WEBSERVICE_ENTRY_ENDPOINT;
+            $webServiceEntryPointIsReachable = $this->checkWebserviceEndpoint($expectedWebserviceEntryPoint);
+
+            $webserviceStatus['endpoint'] = $expectedWebserviceEntryPoint;
+            $webserviceStatus['isFunctional'] = $webServiceEntryPointIsReachable;
+        }
+
         return $this->render(
             '@PrestaShop/Admin/Configure/AdvancedParameters/Webservice/index.html.twig',
             [
                 'help_link' => $this->generateSidebarLink($request->get('_legacy_controller')),
                 'form' => $form->createView(),
                 'grid' => $presentedGrid,
                 'configurationWarnings' => $configurationWarnings,
+                'webserviceStatus' => $webserviceStatus,
             ]
         );
     }
 
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    private function checkWebserviceEndpoint($url): bool
+    {
+        if ($url === null) {
+            return false;
+        }
+
+        $ch = curl_init($url);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 5);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        $data = curl_exec($ch);
+        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+
+        if ($httpCode >= 200 && $httpCode < 300) {
```

Comment:
Why not using Syfmony classes for http codes?

Reply:
I wrote it 😄
https://github.com/PrestaShop/PrestaShop/pull/19335#issuecomment-633568086

As I have not been provided feedback by PM on this new (and unscoped) feature I wait to get approval/specs before doing the polishing. Because it's useless to make the code spotless if I have to redo it 100% because PM ask me to do something different

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/19335/commits/db6866eec2cf932f050ef74981bd353c4e967f52